lib/widgets/profile.dart

```diff
@@ -0,0 +1,334 @@
+              0: FixedColumnWidth(116),
+              1: FlexColumnWidth()},
+            defaultVerticalAlignment: TableCellVerticalAlignment.top,
+            children: displayEntries.map((entry) => TableRow(
+              children: [
+                Padding(
+                  padding: const EdgeInsets.fromLTRB(0, 4, 8, 4),
+                  child: Text(entry.label, style: const TextStyle(fontWeight: FontWeight.bold, ))),
+                ProfileDataTableEntry(entry:entry),
+              ])).toList()),
+        ]));
+  }
+}
+
+List<CustomProfileFieldDisplayEntry> sortAndAnnotateUserFields(Map<int, ProfileFieldUserData> profileData, List<CustomProfileField> customProfileFields) {
+  // TODO(server): The realm-wide field objects have an `order` property,
+  //   but the actual API appears to be that the fields should be shown in
+  //   the order they appear in the array (`custom_profile_fields` in the
+  //   API; our `realmFields` array here.)  See chat thread:
+  //     https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1382982
+  //
+  // We go on to put at the start of the list any fields that are marked for
+  // displaying in the "profile summary".  (Possibly they should be at the
+  // start of the list in the first place, but make sure just in case.)
+  final displayFields = customProfileFields.where((e) => e.displayInProfileSummary == true);
+  final nonDisplayFields = customProfileFields.where((e) => e.displayInProfileSummary != true);
+  return displayFields.followedBy(nonDisplayFields)
+    .where((e) => profileData.containsKey(e.id))
+    .map((e) {
+      final profileField = profileData[e.id]!;
+      return CustomProfileFieldDisplayEntry(label: e.name, text: profileField.value, type: e.type, fieldData: e.fieldData);
+    }).toList();
+}
+
+class ProfileDataTableEntry extends StatelessWidget {
+  const ProfileDataTableEntry({super.key, required this.entry});
+
+  final CustomProfileFieldDisplayEntry entry;
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    // In general this part of the API is not documented up to Zulip's
+    // normal standards.  Some discussion here:
+    //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1387379
+    switch (entry.type) {
+      case CustomProfileFieldType.link:
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: ProfileDataTableLink(url: entry.text, label: entry.text));
+      case CustomProfileFieldType.choice:
+        // TODO(server): This isn't really documented.  But see chat thread:
+        //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1383005
+        final Map<String, dynamic> fieldData = jsonDecode(entry.fieldData);
+        return Padding(
+          padding: const EdgeInsets.all(4),
+          child: Text(fieldData[entry.text]["text"]));
```

Comment:
The code in this `switch` is doing a fair bit of mixing of UI with parsing/data logic. I'd like to separate those — in particular one consequence of mixing them is that the UI choice of "pad text values with `EdgeInsets.all(4)`" is getting repeated in a lot of places.

The way that's done in the zulip-mobile implementation is that the code corresponding to this `switch`, in `interpretCustomProfileField`, returns a small object that's still in the nature of data, and represents a digested form of the data in the profile field:
```js
/** A user's value for a custom profile field, put into meaningful form. */
export type CustomProfileFieldValue =
  | { +displayType: 'text', +text: string }
  | { +displayType: 'link', +text: string, +url: void | URL }
  | { +displayType: 'users', +userIds: $ReadOnlyArray<UserId> };
```
Then the UI code consumes that. The Dart translation of that approach would be to have a small family of data classes, with a base class like `CustomProfileFieldValue` and subclasses `CustomProfileFieldTextValue`, `CustomProfileFieldLinkValue`, `CustomProfileFieldUsersValue`.

An alternate solution that would still achieve separation of UI and data logic would be to give this class helper methods like `_buildText`, `_buildLink`, `_buildUsers`, and have the switch cases call those to construct the widgets.